### ISSUE TITLE
vector of uint16_t template error (C#)

### ISSUE THREAD

**Issue by rwstoneback** [swig/swig#220]
I have the following code:

```cpp
namespace TestNamespace
{
    typedef std::vector<std::uint8_t> Bytes;
}
```

In my interface file, I do:
```cpp
namespace std
{
    %template(Bytes) vector<uint8_t>;
}
```

In the generated **Bytes.cs** file, it correctly uses the C# **byte** type, but everywhere in that file where it should use the Bytes class itself, such as the copy constructor, AddRange, and InsertRange functions, it instead looks like this:

```cpp
public Bytes(SWIGTYPE_p_std__vectorT_unsigned_short_t other) : this(msclPINVOKE.new_Bytes__SWIG_1(SWIGTYPE_p_std__vectorT_unsigned_short_t.getCPtr(other)), true)

public void AddRange(SWIGTYPE_p_std__vectorT_unsigned_short_t values)

public void InsertRange(int index, SWIGTYPE_p_std__vectorT_unsigned_short_t values)
```

**Comment by ezralanglois** [comment]
I made a pull request that may fix this issue for you:
https://github.com/swig/swig/pull/536

Also, did you include %include <stdint.i>?

There are additional problems if you try to use a 64-bit type under Mono on Linux, I plan to make a pull request to fix that problem too.

**Comment by rwstoneback** [comment]
Sorry...I forgot I opened this. I actually think this was fixed somewhere between SWIG versions 3.0.2 and 3.0.6.
